### ISSUE TITLE
suggestions for add  assert_all_requests_are_fired

### ISSUE THREAD

**Issue by leftvalue** [Colin-b/pytest_httpx#15]
good job!👍🏻
I realy like the project and it dost help me a lot.
in some test I use pytest_httpx to mock some requests but I do not need to ensure all requests will be sent
so , perhaps it's a good feature to add a toggle **assert_all_requests_are_fired** just like what **responses** do?

**Comment by Colin-b** [comment]
Thx, sure it can be done. May I know what is the use case for locking more responses than what your application actually sent ? It will help to document the use cases for this feature.

Thx

**Comment by Colin-b** [comment]
Hi @leftvalue, I can put this feature quite fast but I would really like to document the use case before doing so. Can you tell me more?

**Comment by pabtecha** [comment]
Hi @Colin-b let me jump in and provide an example where I find this useful. It is mostly to reduce duplicated code during testing.

Let's say I have a method:

```
import httpx

def foobar():
    with httpx.Client() as client:
        response1 = client.get("http://test_url")
        if response1.json()['condition']:
            raise SomeException
        response2 = client.get("http://test_url2")
    	return response2.json()
```
right now to test this you would do something like:

```
def test_foobar_exception(http_mock: HttpMock):
	http_mock.add_response(method='GET', url='http://test_url', json={'condition': True})
	
	with pytest.raises(SomeException):
		foobar()

def test_foobar(http_mock: HttpMock):
	http_mock.add_response(method='GET', url='http://test_url', json={'condition': False})
	http_mock.add_response(method='GET', url='http://test_url2', json={'foo': 'bar'})
	
	assert foobar() == {'foo': 'bar'}
```

It would be nice to be able to mock all the possible responses and then let the developer check if it was called or not. That way you could reuse some code and do something like:

```
@contextmanager
def mock_requests(http_mock: HttpMock, condition: bool = False):
	http_mock.add_response(method='GET', url='http://test_url', json={'condition': condition})
	http_mock.add_response(method='GET', url='http://test_url2', json={'foo': 'bar'})
	yield

def test_foobar_exception(http_mock: HttpMock):
	http_mock.add_response(method='GET', url='http://test_url', json={'condition': True})
	
	with pytest.raises(SomeException):
		with mock_requests(http_mock, condition=True):
			foobar()

def test_foobar(http_mock: HttpMock):
	with mock_requests(http_mock, condition=False):
		assert foobar() == {'foo': 'bar'}
``` 

It is specially useful when the responses are big/ there are several endpoints.

I hope it helps to illustrate some cases when someone might want not check that all requests were called.

**Comment by Colin-b** [comment]
I assume `test_foobar_exception` was actually 

```python
def test_foobar_exception(http_mock: HttpMock):
	with pytest.raises(SomeException):
		with mock_requests(http_mock, condition=True):
			foobar()
```

in your example.

I will add the option then (but you can come up with a PR if you want)

**Comment by leftvalue** [comment]
> Thx, sure it can be done. May I know what is the use case for mocking more responses than what your application actually sent ? It will help to document the use cases for this feature.
> 
> Thx

think about that:
in function a, I send a request named A,
in function b, I send a request named B,
in function c,  I send a request named C

then it's an interesting example
function d call func a & b
function e call func b & c
function f call func a & c

you can think about I wanna share all request mock in a big register,so that all function can share the mock, but any function have no chance to send all mocked requests inner the mock request register at all.

**Comment by leftvalue** [comment]
> 2\. fixture

I think both of it is quite great!

**Comment by Colin-b** [comment]
Hello,

Apologies for the delay, I am still under quite a bit of load but I managed some time to implement the fixture (as a start).

Next release will contains the `assert_all_responses_were_requested` fixture:

```python
import pytest

@pytest.fixture
def assert_all_responses_were_requested() -> bool:
    return False
```

**Comment by Colin-b** [comment]
Fixed in release 0.5.0
